### DISCUSSION TITLE
Deleted Storage objects are still found in `storage.objects`?

### DISCUSSION THREAD

**Discussion by liaujianjie** [supabase/supabase#2559] [Questions]
When Storage objects are deleted, their rows can still be found by querying.

```sql
select * from storage.objects;
```

Observing this behaviour, I have a few questions:
- What's the reason for this?
- Is it safe to manually remove these deleted objects with `DELETE FROM storage.objects ...`?
- What happens if I execute `DELETE FROM storage.objects ...` on undeleted objects?
- When a Storage file is deleted (from the dashboard, isomorphic JS client, or CLI), are they actually deleted from the underlying S3 bucket? (Not that it matters to me functionally, I'm just curious.)

#### Edit:

I just realised my observation was wrong because of a bug. The rows in `storage.objects` are getting deleted as expected when I delete individual files (tested on the dashboard).

What was happening instead was that I tried to delete files by a "folder", which shows up as a successful deletion even though nothing was deleted. I'll file an issue for this bug.

**Comment by silentworks** [comment]
I think there is a release coming shortly which should fix the deleting by folder issue, but file the issue anyway. I think @inian will be able to answer if the upcoming release fixes this.

**Reply by chrisLoPresti** [reply]
just confirming im seeing the same problem
